packages/stark-ui/src/modules/message-pane/services/message-pane.service.spec.ts

```diff
@@ -0,0 +1,210 @@
+			const mockMessages: StarkMessage[] = [mockInfoMessage, mockWarningMessage, mockErrorMessage];
+
+			messagePaneService.add(mockMessages);
+
+			mockMessages[0] = { ...mockInfoMessage, priority: 999 };
+			mockMessages[1] = { ...mockWarningMessage, priority: 999 };
+			mockMessages[2] = { ...mockErrorMessage, priority: 999 };
+
+			expect(mockStore.dispatch).toHaveBeenCalledTimes(1);
+			expect(mockStore.dispatch).toHaveBeenCalledWith(new StarkAddMessages(mockMessages));
+		});
+	});
+
+	describe("addOne", () => {
+		it("should dispatch the GET_ALL_MESSAGES action and return the observable used to fetch the latest value of the starkMessages", () => {
+			messagePaneService.addOne(mockInfoMessage);
+			const validatedMessage: StarkMessage = { ...mockInfoMessage, priority: 999 };
+
+			expect(mockStore.dispatch).toHaveBeenCalledTimes(1);
+			expect(mockStore.dispatch).toHaveBeenCalledWith(new StarkAddMessages([validatedMessage]));
+		});
+
+		it("should add the default priority of 999, when it is not supplied", () => {
+			const message: StarkMessage = {
+				id: "3",
+				key: "MESSAGES.INFOS.DUMMY",
+				interpolateValues: { var1: "some info value", var2: "whatever" },
+				code: "1234",
+				type: StarkMessageType.INFO
+			};
+
+			messagePaneService.addOne(message);
+			const validatedMessage: StarkMessage = { ...message, priority: 999 };
+			expect(mockStore.dispatch).toHaveBeenCalledWith(new StarkAddMessages([validatedMessage]));
+		});
+
+		it("should throw an error when the priority is smaller than 1", () => {
+			const message: StarkMessage = {
+				id: "3",
+				key: "MESSAGES.INFOS.DUMMY",
+				interpolateValues: { var1: "some info value", var2: "whatever" },
+				code: "1234",
+				type: StarkMessageType.INFO,
+				priority: -1
+			};
+
+			expect(() => messagePaneService.addOne(message)).toThrowError(/priority has to be between 1 and 999/);
+		});
+
+		it("should throw an error when the priority is greater than 999", () => {
+			const message: StarkMessage = {
+				id: "3",
+				key: "MESSAGES.INFOS.DUMMY",
+				interpolateValues: { var1: "some info value", var2: "whatever" },
+				code: "1234",
+				type: StarkMessageType.INFO,
+				priority: 1000
+			};
+
+			expect(() => messagePaneService.addOne(message)).toThrowError(/priority has to be between 1 and 999/);
+		});
+
+		it("should not change a priority between 1 and 999", () => {
+			const message: StarkMessage = new StarkMessageImpl();
+			message.id = "3";
+			message.key = "MESSAGES.INFOS.DUMMY";
+			message.interpolateValues = { var1: "some info value", var2: "whatever" };
+			message.code = "1234";
+			message.type = StarkMessageType.INFO;
+			message.priority = 200;
+
+			messagePaneService.addOne(message);
+			expect(mockStore.dispatch).toHaveBeenCalledWith(new StarkAddMessages([message]));
+		});
+	});
+
+	describe("getAll", () => {
+		it("should dispatch the ADD_MESSAGES action passing the given messages in the payload", () => {
```

Comment:
"should dispatch the ADD_MESSAGES action..." -> "should dispatch the GET_ALL_MESSAGES action..."